### PR TITLE
UnitOfWork::clear() misses $eagerLoadingEntities

### PR DESCRIPTION
**Pull request by BenMorel**
AFAICS, eager loading entities should be cleared as well when the UnitOfWork is cleared.

Actually, this is [already the case](https://github.com/doctrine/orm/blob/master/lib/Doctrine/ORM/UnitOfWork.php#L1982) on `master`!

### CODE REVIEW

**Comment by greg0ire** [comment]
`phpcs` has a weird complaint I don't understand… can you try running `vendor/bin/phpcbf`?

**Comment by BenMorel** [comment]
I'm not sure why this happens either... Locally, `phpcs` reports, and `phpcbf` fixes, thousands (!) of CS issues.

I'm not familiar with `phpcs`, but I suspect from the command:

    php git-phpcs.phar origin/$TRAVIS_BRANCH...$TRAVIS_PULL_REQUEST_SHA

that it only reports issues on the line(s) modified by the PR. And due to the surrounding lines, there is not much I can do to fix this.

**Comment by greg0ire** [comment]
Maybe you could try aligning all equal signs, including in the surrounding lines?

**Comment by BenMorel** [comment]
Let's try this!

**Comment by greg0ire** [comment]
Hey! That actually worked! :)

**Comment by greg0ire** [comment]
Original commit: d070fc62a76cfe7b9ae35c3236335ab888762833

**Review by greg0ire — DISMISSED** [review]

**Comment by greg0ire** [comment]
The commit above contains a similar change for `$persisters`, should it be applied too?

**Comment by BenMorel** [comment]
I have unfortunately no idea. For sure anything that contains a reference to an entity must be cleared, hence this PR, for the rest I'm not the right person to ask ;)

**Comment by greg0ire** [comment]
Let's wait for feedback from contributors that know the UOW better than I do.

**Review by lcobucci — CHANGES_REQUESTED** [review]
@BenMorel thanks for the patch. The fix is 100% valid but we need a test to be able to get it merged :+1:

**Comment by BenMorel** [comment]
@lcobucci Adding a test does not look like a trivial task, as there is no setter/getter for `$eagerLoadingEntities`.

Suggestions:

- test using reflection 🤢
- side-test using `triggerEagerLoads()`; this would require a complex setting, where you have to somehow set up an eager loading entity, and check that its `EntityPersister` is not invoked when calling `triggerEagerLoads()` after `clear()`; *really* not trivial, in comparison to the fix!

**Comment by SenseException** [comment]
Suggestion: You could use an EntityManager in a functional test to setup eagerloading entities and test the clear-method of the UOW.

**Comment by ostrolucky** [comment]
Test case added

**Comment by BenMorel** [comment]
@ostrolucky Thank you!

**Review by greg0ire — APPROVED** [review]

**Review by guilhermeblanco — APPROVED** [review]
This was an issue I found when i was working on ephemeral unit of work support. It also helps on freeing resources as eager loaded entities hang there forever. Patch on its own is great.

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH7869Test.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+use Doctrine\Tests\Mocks\DriverMock;
+use Doctrine\Tests\Mocks\EntityManagerMock;
+use Doctrine\Tests\OrmTestCase;
+
+/**
+ * @group GH7869
+ */
+class GH7869Test extends OrmTestCase
+{
+    public function testDQLDeferredEagerLoad()
```
Comment (ostrolucky) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH7869Test.php]: I forgot to rename test case 😞 Let me know if I Should create new PR with just this change

**Pull request merged**